### PR TITLE
20200210 cosmetic

### PR DESCRIPTION
**Pull request by lyakh**
Several cosmetic patches

### CODE REVIEW

File: sound/soc/intel/skylake/skl.c [hunk 1]
```diff
@@ -794,7 +794,7 @@ static void skl_probe_work(struct work_struct *work)
 {
 	struct skl_dev *skl = container_of(work, struct skl_dev, probe_work);
 	struct hdac_bus *bus = skl_to_bus(skl);
-	struct hdac_ext_link *hlink = NULL;
+	struct hdac_ext_link *hlink;
```
Comment (plbossart) [inline sound/soc/intel/skylake/skl.c:797]: need to split those patches in two

ASoC: codecs: hdac_hdmi: 

and 

ASoC: Intel: Skylake:...

File: sound/soc/intel/boards/sof_pcm512x.c [hunk 2]
```diff
@@ -383,7 +383,7 @@ static int sof_audio_probe(struct platform_device *pdev)
 	INIT_LIST_HEAD(&ctx->hdmi_pcm_list);
 
 	sof_audio_card_pcm512x.dev = &pdev->dev;
-	mach = (&pdev->dev)->platform_data;
+	mach = pdev->dev.platform_data;
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_pcm512x.c]: it may be clumsy but that's what we use in all other machine drivers. Either we change them all or we don't

File: sound/soc/sof/intel/hda.c [hunk 3]
```diff
@@ -1026,7 +1025,8 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 
 	/* used by hda machine driver to create dai links */
 	if (pdata->machine) {
-		mach_params = (struct snd_soc_acpi_mach_params *)
+		struct snd_soc_acpi_mach_params *mach_params =
+			(struct snd_soc_acpi_mach_params *)
```
Comment (plbossart) [inline sound/soc/sof/intel/hda.c]: I am not a big fan of this, we don't do it anywhere else.

Comment (ranj063) [reply sound/soc/sof/intel/hda.c]: +1 I dont see any added value in reducing the scope for the variable in this function too.

Comment (lyakh) [reply sound/soc/sof/intel/hda.c]: @plbossart @ranj063 In this specific function moving just one variable inside the block where it is used might not make a huge difference, sure. It's "just" one variable - it's easy enough to check where it's used. But in general you'll probably agree, that it is a good practice to limit the scope of variables to the smallest code block, where they are used. Firstly not to clutter the function scope, i.e. to make it easier to understand the function as a whole, to see which variables are used there. Secondly to easily see where that specific variable is needed. Putting it inside that block immediately tells the reader, that that variable isn't used anywhere outside of it. Both these aspects help readability, I think, and should be encouraged. There are also edge cases like
```
static void f(void)
{
	int i;
	if (x) {
		for (i = 0; i < N; i++) {}
	}
	if (y) {
		for (i = 0; i < N; i++) {}
	}
}
```
In the above case the variable `i` is only needed in respective `if` clauses and it doesn't have to carry its value from one clause to another, so you can define it within each `if` clause. OTOH if the function is simple enough, doesn't have many variables defined in its scope, and if `i` maybe even used more than twice - you might as well save defining it in each enclosing block and just define it once for the whole function.
In short - this isn't something important, but in general, especially where there are many variables, that make the function header difficult to grasp, and if some of those variables clearly are only needed in some limited scope within the function, it's better to have their definitions limited to that scope.

File: sound/soc/intel/skylake/skl.c (hunk 1 above)
Comment (lyakh) [reply sound/soc/intel/skylake/skl.c:797]: done

File: sound/soc/intel/boards/sof_pcm512x.c (hunk 2 above)
Comment (lyakh) [reply sound/soc/intel/boards/sof_pcm512x.c]: done

File: sound/soc/sof/intel/hda.c (hunk 3 above)
Comment (lyakh) [reply sound/soc/sof/intel/hda.c]: I've dropped this one for now, let me know if it's wanted after all

File: sound/soc/intel/boards/bxt_wm8804.c [hunk 4]
```diff
@@ -225,7 +225,7 @@ static int bxt_wm8804_probe(struct platform_device *pdev)
 	int ret_val = 0;
 	int i;
 
-	mach = (&pdev->dev)->platform_data;
+	mach = pdev->dev.platform_data;
```
Comment (plbossart) [inline sound/soc/intel/boards/bxt_wm8804.c:228]: would you mind putting this in a separate patch, this machine driver is not upstream so it'll require me or Ranjani to split it later. Thanks!

File: sound/soc/intel/boards/sdw_rt700.c [hunk 5]
```diff
@@ -279,7 +279,7 @@ static int mc_probe(struct platform_device *pdev)
 	card->dev = &pdev->dev;
 
 	/* override platform name, if required */
-	mach = (&pdev->dev)->platform_data;
+	mach = pdev->dev.platform_data;
```
Comment (plbossart) [inline sound/soc/intel/boards/sdw_rt700.c:282]: same for all sdw parts, not upstream just yet. One patch per non-upstream driver would be real nice.

**Comment by lyakh** [comment]
> looks good but could you split the 2nd patch between what's upstream and not, it'll help us manage out-of-tree patches in an easier way. Thanks!

@plbossart "upstream" means Mark's tree, right?

File: sound/soc/intel/boards/bxt_wm8804.c (hunk 4 above)
Comment (ranj063) [reply sound/soc/intel/boards/bxt_wm8804.c:228]: @plbossart havent we made such changes to machine drivers, in a separate patch per machine driver before?

File: sound/soc/intel/skylake/skl.c [hunk 6]
```diff
@@ -359,7 +359,7 @@ static int skl_resume(struct device *dev)
 	struct pci_dev *pci = to_pci_dev(dev);
 	struct hdac_bus *bus = pci_get_drvdata(pci);
 	struct skl_dev *skl  = bus_to_skl(bus);
-	struct hdac_ext_link *hlink = NULL;
```
Comment (ranj063) [inline sound/soc/intel/skylake/skl.c:362]: @lyakh can we drop the "(cosmetic)" in the commit title?

Comment (lyakh) [reply sound/soc/intel/skylake/skl.c:362]: @ranj063 sorry, don't understand, can you explain why in this specific patch? Or in all patches? I think it's perfectly appropriate, since these patches don't fix any bugs or checkpatch or compiler warnings, nor do they implement any features. They are cosmetic.

Comment (ranj063) [reply sound/soc/intel/skylake/skl.c:362]: @lyakh sorry just realized that this is there in all patches. Absolutely agree this is cosmetic and I thought its quite obvious, hence the suggestion. But up to you if you want to keep it.

**Comment by plbossart** [comment]
> > looks good but could you split the 2nd patch between what's upstream and not, it'll help us manage out-of-tree patches in an easier way. Thanks!
> 
> @plbossart "upstream" means Mark's tree, right?

yes indeed.

File: sound/soc/intel/boards/bxt_wm8804.c (hunk 4 above)
Comment (plbossart) [reply sound/soc/intel/boards/bxt_wm8804.c:228]: I prefer one patch per machine driver when we change functionality so that bisections are easier but here it's fine.

File: sound/soc/sof/intel/hda.c (hunk 3 above)
Comment (kv2019i) [reply sound/soc/sof/intel/hda.c]: @lyakh @plbossart @ranj063 This is interesting. I was under (false) impression that C99 style declarations are not used at least in ALSA, but upon checking the latest codebase, that's not really the case. In ASoC as well, we have C99 style declarations now, e.g. in soc-core.c:snd_soc_add_controls() and snd-pcm.c:soc_new_pcm(). Of course, in SOF we don't have them, so maybe better stick to existing style unless with a very good reason, but at least something to consider. I'm more used to C99 style local declarations myself, but I admit the older kernel coding style of keeping functions short, helps with many of the same issues (i.e. if you have more than 6-7 local variables in your function, you are doing something wrong).

Comment (lyakh) [reply sound/soc/sof/intel/hda.c]: @kv2019i sorry, which C99 style declarations do you mean? If you're talking about declaring variables at the block ("compound statement" K&R) level, that's ANSI C, it's also present in K&R. I'm not proposing `for (int i = 0;...)`

Comment (kv2019i) [reply sound/soc/sof/intel/hda.c]: @lyakh That's good a clarification. C99 relaxed the rules, so I tend to just refer to it when talking about more C++ like style to introduce variables close to place of use, but you are 100% right declarations at start of compounds have been valid from the start. It just has been a bit error prone as gcc has been more relaxed already before C99, so if you used local variable in middle of functions, and gcc, it was easy to introduce gcc-extension'isms to your code. But in kernel, I did not find any examples of real C99 style local variables. That might still be frowned upon.

Comment (plbossart) [reply sound/soc/sof/intel/hda.c]: let's stick to traditional kernel style, we have more pressing things to argue about

**Comment by plbossart** [comment]
@lyakh 2 conflicts need to be resolved, thanks!

File: sound/soc/sof/intel/hda.c (hunk 3 above)
Comment (lyakh) [reply sound/soc/sof/intel/hda.c]: @plbossart I have dropped this patch upon popular request, but I'm pretty sure this isn't working against the traditional kernel style. I don't remember *ever* seeing this kind of code rejected on grounds of style. Of course the kernel consists of many subsystems and each subsystem has its own preferences / traditions, and I still don't know ALSA preferences well enough to argue about them specifically. If this is indeed the preferred style within ALSA or ASoC, please let me know!

Comment (paulstelian97) [reply sound/soc/sof/intel/hda.c]: In the kernel there are two major differing styles with small differences, namely networking and everything else.
